rdrf/rdrf/services/rest/views/proms_api.py

```diff
@@ -45,13 +57,9 @@ def post(self, request):
         survey_name = data.get("survey_name")
 
         try:
-            registry_model = Registry.objects.get(code=registry_code)
-
-            survey_model = Survey.objects.get(registry=registry_model,
-                                              name=survey_name)
-
-            survey_assignment = SurveyAssignment.objects.get(registry=survey_model.registry,
-                                                             survey_name=survey_model.name,
+            registry_model = get_registry(registry_code)
```

Comment:
according to this https://docs.djangoproject.com/en/3.2/topics/db/queries/#queries-over-related-objects
we only need primary key 
so maybe get id of registry from code :
id = Registry.objects.get(code=code).values('id')

then

survey_assignment = SurveyAssignment.objects.get(registry=id

Reply:
.values cannot be used with .get - can be applied to queryset

Reply:
Registry.objects.filter(code="ICHOMCRC").values_list('id', flat=True)[0]